data-prepper-plugins/mutate-event-processors/src/test/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorTests.java

```diff
@@ -441,21 +441,113 @@ public void testMetadataKeySetWithFormatOverwriteExistingEntry() {
 
     @Test
     public void testMetadataKeyAndKeyBothNotSetThrows() {
-        assertThrows(IllegalArgumentException.class, () -> createEntry(null, null, "newValue", null, false, null));
+        assertThrows(IllegalArgumentException.class, () -> createEntry(null, null, "newValue", null, null, false, null));
     }
 
     @Test
     public void testMetadataKeyAndKeyBothSetThrows() {
-        assertThrows(IllegalArgumentException.class, () -> createEntry("newKey", "newMetadataKey", "newValue", null, false, null));
+        assertThrows(IllegalArgumentException.class, () -> createEntry("newKey", "newMetadataKey", "newValue", null, null, false, null));
+    }
+
+    @Test
+    public void testOnlyOneTypeOfValueIsSupported() {
+        assertThrows(RuntimeException.class, () -> createEntry("newKey", "newMetadataKey", "newValue", "/newFormat", null, false, null));
+    }
+
+    @Test
+    public void testOnlyOneTypeOfValueIsSupportedWithExpressionAndFormat() {
+        assertThrows(RuntimeException.class, () -> createEntry("newKey", "newMetadataKey", null, "/newFormat", "length(/message)", false, null));
+    }
+
+    @Test
+    public void testOnlyOneTypeOfValueIsSupportedWithValueAndExpressionAndFormat() {
+        assertThrows(RuntimeException.class, () -> createEntry("newKey", "newMetadataKey", "value", "/newFormat", "length(/message)", false, null));
+    }
+
+    @Test
+    public void testWithAllValuesNull() {
+        assertThrows(RuntimeException.class, () -> createEntry("newKey", "newMetadataKey", null, null, null, false, null));
+    }
+
+    @Test
+    public void testValueExpressionWithArithmeticExpression() {
+        String valueExpression = "/number-key";
+        when(mockConfig.getEntries()).thenReturn(createListOfEntries(createEntry("num_key", null, null, null, valueExpression, false, null)));
+        final AddEntryProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getTestEventWithMultipleDataTypes();
+        when(expressionEvaluator.evaluate(valueExpression, record.getData())).thenReturn(record.getData().get(valueExpression, Integer.class));
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+        Event event = editedRecords.get(0).getData();
+        assertThat(event.get("num_key", Integer.class), equalTo(1));
+    }
+
+    @Test
+    public void testValueExpressionWithStringExpression() {
+        String valueExpression = "/string-key";
+        when(mockConfig.getEntries()).thenReturn(createListOfEntries(createEntry("num_key", null, null, null, valueExpression, false, null)));
+        final AddEntryProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getTestEventWithMultipleDataTypes();
+        when(expressionEvaluator.evaluate(valueExpression, record.getData())).thenReturn(record.getData().get(valueExpression, String.class));
```

Comment:
From a testing perspective, we just want to get the value from `expressionEvaluator`.

I think you could replace this: `record.getData().get(valueExpression, String.class)` with some arbitrary string value.

```
String evaluatedValue = UUID.randomUUID().toString();
when(expressionEvaluator.evaluate(valueExpression, record.getData())).thenReturn(evaluatedValue);
```

Reply:
How is this more accurate? You have mocked this to return some value from inside the Event data. There is some logic in the tests that make them more difficult to understand and more coupled to previous sets on the event model.